src/hooks/useEvidences.js

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from 'react'
+import { ipfsGet } from '../lib/ipfs-utils'
+import isIPFS from 'is-ipfs'
+import { ERROR_TYPES } from '../types/evidences-status-types'
+
+export default function useEvidences(evidences) {
+  const [fetchedData, setFetchedData] = useState()
+
+  useEffect(() => {
+    async function getEvidenceData() {
+      const evidenceFetchedData = await Promise.all(
+        evidences.map(async evidence => {
+          const { data: evidenceData, submitter, createdAt } = evidence
+
+          // check if the metadata inside the evidence is a cid
+          if (isIPFS.multihash(evidenceData) || isIPFS.cid(evidenceData)) {
+            const { data, error } = await ipfsGet(evidenceData)
+            if (!error) {
+              if (typeof data === 'object') {
+                // we are just interested in those keys
+                return {
+                  metadata: data.metadata || '',
```

Comment:
It'd be great to add these (for now) to a `docs/metadata.md` file to keep track of what keys we're assuming.